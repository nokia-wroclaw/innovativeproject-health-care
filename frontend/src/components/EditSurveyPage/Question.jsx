import React from 'react';
import { Segment, Header } from 'semantic-ui-react';

const Question = ({ value }) => {
  return (
    <Segment>
      <Header as='h4'>{value}</Header>
    </Segment>
  );
};

export default Question;
