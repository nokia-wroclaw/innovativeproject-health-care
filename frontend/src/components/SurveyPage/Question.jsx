import React from 'react';
import { Grid, Header } from 'semantic-ui-react';
import ColoredCheckbox from './ColoredCheckbox';

const Question = ({ question }) => {
  return (
    <Grid>
      <Grid.Column mobile={10} tablet={13} computer={13}>
        <Header as='h5'>{question.value}</Header>
      </Grid.Column>
      <Grid.Column mobile={6} tablet={3} computer={3}>
        <div className='flex-space-evenly'>
          <ColoredCheckbox
            color='green'
            value={2}
            question={question}
            hintText="It's OK"
          />
          <ColoredCheckbox
            color='yellow'
            value={1}
            question={question}
            hintText="It's not bad but could be better"
          />
          <ColoredCheckbox
            color='red'
            value={0}
            question={question}
            hintText="It's really bad"
          />
        </div>
      </Grid.Column>
    </Grid>
  );
};

export default Question;
