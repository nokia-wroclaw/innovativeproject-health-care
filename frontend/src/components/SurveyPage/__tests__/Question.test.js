import React from "react";
import { shallow } from "enzyme";
import Question from "../Question";

describe("Question", () => {
  it("should match snapshot", () => {
    expect(shallow(<Question question={{}} />)).toMatchSnapshot();
  });
});
