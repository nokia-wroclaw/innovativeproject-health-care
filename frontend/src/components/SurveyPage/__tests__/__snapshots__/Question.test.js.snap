// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Question
    question={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <GridColumn
          computer={13}
          mobile={10}
          tablet={13}
        >
          <Header
            as="h5"
            style={
              Object {
                "marginBottom": 0,
              }
            }
          />
          <span />
          <span
            style={
              Object {
                "color": "rgb(235, 40, 40)",
                "fontSize": "x-large",
              }
            }
          >
            
          </span>
        </GridColumn>,
        <GridColumn
          computer={3}
          mobile={6}
          tablet={3}
        >
          <div
            className="flex-space-evenly"
          >
            <Connect(ColoredCheckbox)
              color="green"
              hintText="It's OK"
              question={Object {}}
              value={2}
            />
            <Connect(ColoredCheckbox)
              color="yellow"
              hintText="It's not bad but could be better"
              question={Object {}}
              value={1}
            />
            <Connect(ColoredCheckbox)
              color="red"
              hintText="It's really bad"
              question={Object {}}
              value={0}
            />
          </div>
        </GridColumn>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Header
              as="h5"
              style={
                Object {
                  "marginBottom": 0,
                }
              }
            />,
            <span />,
            <span
              style={
                Object {
                  "color": "rgb(235, 40, 40)",
                  "fontSize": "x-large",
                }
              }
            >
              
            </span>,
          ],
          "computer": 13,
          "mobile": 10,
          "tablet": 13,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "h5",
              "children": undefined,
              "style": Object {
                "marginBottom": 0,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "style": Object {
                "color": "rgb(235, 40, 40)",
                "fontSize": "x-large",
              },
            },
            "ref": null,
            "rendered": "",
            "type": "span",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            className="flex-space-evenly"
          >
            <Connect(ColoredCheckbox)
              color="green"
              hintText="It's OK"
              question={Object {}}
              value={2}
            />
            <Connect(ColoredCheckbox)
              color="yellow"
              hintText="It's not bad but could be better"
              question={Object {}}
              value={1}
            />
            <Connect(ColoredCheckbox)
              color="red"
              hintText="It's really bad"
              question={Object {}}
              value={0}
            />
          </div>,
          "computer": 3,
          "mobile": 6,
          "tablet": 3,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(ColoredCheckbox)
                color="green"
                hintText="It's OK"
                question={Object {}}
                value={2}
              />,
              <Connect(ColoredCheckbox)
                color="yellow"
                hintText="It's not bad but could be better"
                question={Object {}}
                value={1}
              />,
              <Connect(ColoredCheckbox)
                color="red"
                hintText="It's really bad"
                question={Object {}}
                value={0}
              />,
            ],
            "className": "flex-space-evenly",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "green",
                "hintText": "It's OK",
                "question": Object {},
                "value": 2,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "yellow",
                "hintText": "It's not bad but could be better",
                "question": Object {},
                "value": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "red",
                "hintText": "It's really bad",
                "question": Object {},
                "value": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <GridColumn
            computer={13}
            mobile={10}
            tablet={13}
          >
            <Header
              as="h5"
              style={
                Object {
                  "marginBottom": 0,
                }
              }
            />
            <span />
            <span
              style={
                Object {
                  "color": "rgb(235, 40, 40)",
                  "fontSize": "x-large",
                }
              }
            >
              
            </span>
          </GridColumn>,
          <GridColumn
            computer={3}
            mobile={6}
            tablet={3}
          >
            <div
              className="flex-space-evenly"
            >
              <Connect(ColoredCheckbox)
                color="green"
                hintText="It's OK"
                question={Object {}}
                value={2}
              />
              <Connect(ColoredCheckbox)
                color="yellow"
                hintText="It's not bad but could be better"
                question={Object {}}
                value={1}
              />
              <Connect(ColoredCheckbox)
                color="red"
                hintText="It's really bad"
                question={Object {}}
                value={0}
              />
            </div>
          </GridColumn>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Header
                as="h5"
                style={
                  Object {
                    "marginBottom": 0,
                  }
                }
              />,
              <span />,
              <span
                style={
                  Object {
                    "color": "rgb(235, 40, 40)",
                    "fontSize": "x-large",
                  }
                }
              >
                
              </span>,
            ],
            "computer": 13,
            "mobile": 10,
            "tablet": 13,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "h5",
                "children": undefined,
                "style": Object {
                  "marginBottom": 0,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "style": Object {
                  "color": "rgb(235, 40, 40)",
                  "fontSize": "x-large",
                },
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div
              className="flex-space-evenly"
            >
              <Connect(ColoredCheckbox)
                color="green"
                hintText="It's OK"
                question={Object {}}
                value={2}
              />
              <Connect(ColoredCheckbox)
                color="yellow"
                hintText="It's not bad but could be better"
                question={Object {}}
                value={1}
              />
              <Connect(ColoredCheckbox)
                color="red"
                hintText="It's really bad"
                question={Object {}}
                value={0}
              />
            </div>,
            "computer": 3,
            "mobile": 6,
            "tablet": 3,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(ColoredCheckbox)
                  color="green"
                  hintText="It's OK"
                  question={Object {}}
                  value={2}
                />,
                <Connect(ColoredCheckbox)
                  color="yellow"
                  hintText="It's not bad but could be better"
                  question={Object {}}
                  value={1}
                />,
                <Connect(ColoredCheckbox)
                  color="red"
                  hintText="It's really bad"
                  question={Object {}}
                  value={0}
                />,
              ],
              "className": "flex-space-evenly",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "color": "green",
                  "hintText": "It's OK",
                  "question": Object {},
                  "value": 2,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "color": "yellow",
                  "hintText": "It's not bad but could be better",
                  "question": Object {},
                  "value": 1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "color": "red",
                  "hintText": "It's really bad",
                  "question": Object {},
                  "value": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
